viewing_party/party.py

```diff
@@ -1,23 +1,352 @@
+    ```python
+    {
+        "title": "Title A",
+        "genre": "Horror",
+        "rating": 3.5
+    }
+    ```
+    add the `movie` to the `"watchlist"` list inside of `user_data`
+    return the `user_data`
+    
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data 
+
+
+def watch_movie(user_data, title):
+    '''
+    take two parameters: `user_data`, `title`
+    the value of `user_data` will be a dictionary with a `"watchlist"`and a `"watched"`
+    This represents that the user has a watchlist and a list of watched movies
+    the value of `title` will be a string
+    This represents the title of the movie the user has watched
+    If the title is in a movie in the user's watchlist:
+    remove that movie from the watchlist
+    add that movie to watched
+    return the `user_data`
+    If the title is not a movie in the user's watchlist:
+    return the `user_data`
+    Note: For Waves 2, 3, 4, and 5, your implementation of each 
+    of the functions should not modify `user_data`.
+'''
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+
+    return user_data
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+# {"watched":[rating: 3.3]}
+def get_watched_avg_rating(user_data):
+    '''
+    take one parameter: `user_data`
+    the value of `user_data` will be a dictionary with a `"watched"` list of movie dictionaries
+    This represents that the user has a list of watched movies
+    Calculate the average rating of all movies in the watched list
+    The average rating of an empty watched list is `0.0`
+    return the average rating
+    
+    '''
+    avgerage_rating = 0
+    total_rating = 0
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            total_rating += movie["rating"]
+        avgerage_rating = total_rating/len(user_data["watched"])
+        return avgerage_rating
+    else: 
+        return 0.0
+
+
+def get_most_watched_genre(user_data):
+    '''
+    take one parameter: `user_data`
+    the value of `user_data` will be a dictionary with a `"watched"` 
+    list of movie dictionaries. Each movie dictionary has a key `"genre"`.
+    This represents that the user has a list of watched movies. 
+    Each watched movie has a genre.
+    The values of `"genre"` is a string.
+    Determine which genre is most frequently occurring in the watched list
+    return the genre that is the most frequently watched
+    If the value of "watched" is an empty list, `get_most_watched_genre` should return `None`.
+    
+    '''
+    #if user's watched list is not empty, we count the number of genre 
+    #watched most frequently  
+    #["horror", "fantasy", "horror"]
+    #{"horror": 2, "fantasy": 1}

```

Comment:
The comments should just be a short comment explaining how your code is working. Any other comments could be added to a .txt file for you to reference later.